dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -0,0 +1,210 @@
+        Kernel kernel,
+        KernelArguments executionContext,
+        IHandlebars handlebarsInstance,
+        KernelFunctionMetadata functionMetadata,
+        string nameDelimiter,
+        CancellationToken cancellationToken = default)
+    {
+        string fullyResolvedFunctionName = functionMetadata.PluginName + nameDelimiter + functionMetadata.Name;
+
+        KernelHelpersUtils.RegisterHelperSafe(
+            handlebarsInstance,
+            fullyResolvedFunctionName,
+            (Context context, Arguments handlebarsArguments) =>
+            {
+                // Get the parameters from the template arguments
+                if (handlebarsArguments.Length is not 0)
+                {
+                    if (handlebarsArguments[0].GetType() == typeof(HashParameterDictionary))
+                    {
+                        ProcessHashArguments(functionMetadata, executionContext, (IDictionary<string, object>)handlebarsArguments[0], nameDelimiter);
+                    }
+                    else
+                    {
+                        ProcessPositionalArguments(functionMetadata, executionContext, handlebarsArguments);
+                    }
+                }
+                else if (functionMetadata.Parameters.Any(p => p.IsRequired))
+                {
+                    throw new KernelException($"Invalid parameter count for function {functionMetadata.Name}. {handlebarsArguments.Length} were specified but {functionMetadata.Parameters.Count} are required.");
+                }
+
+                KernelFunction function = kernel.Plugins.GetFunction(functionMetadata.PluginName, functionMetadata.Name);
+
+                // Invoke the function and write the result to the template
+                return InvokeKernelFunction(kernel, function, executionContext, cancellationToken);
+            });
+    }
+
+    /// <summary>
+    /// Checks if handlebars argument is a valid type for the function parameter.
+    /// Must satisfy one of the following:
+    /// Types are an exact match.
+    /// Argument is any kind of numeric type if function parameter requires a numeric type.
+    /// Argument type is an object (this covers complex types).
+    /// Function parameter is a generic type.
+    /// </summary>
+    /// <param name="parameterType">Function parameter type.</param>
+    /// <param name="argument">Handlebar argument.</param>
+    private static bool IsExpectedParameterType(KernelParameterMetadata parameterType, object argument)
+    {
+        var actualParameterType = parameterType.ParameterType.TryGetGenericNullableType(out var nullableType) ? nullableType : parameterType.ParameterType;
+
+        bool parameterIsNumeric = KernelHelpersUtils.IsNumericType(actualParameterType)
+            || (parameterType.Schema?.RootElement.TryGetProperty("type", out JsonElement typeProperty) == true && typeProperty.GetString() == "number");
+
+        bool argIsNumeric = KernelHelpersUtils.IsNumericType(argument?.GetType())
```

Comment:
The `?` in the call `argument?.GetType()` assumes that the argument is nullable, but the method signature does not indicate this.

Reply:
removed the `?`